src/Pages/Main/Components/FoodMap/Filter/StoreOption/StoreOption.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+
+const StoreOption = ({ getCategoryCheck }) => {
+  const [selectCategory, setSelectCategory] = useState(
+    Array(1).fill(true).concat(Array(11).fill(false))
+  );
+  const checkCategory = e => {
+    const {
+      dataset: { target },
+    } = e.target;
+
+    const saveType = [
+      selectCategory.map((category, index) =>
+        index === 0 ? !category : false
+      ),
+      selectCategory.map((category, index) =>
+        index === 0
+          ? false
+          : index + 1 === Number(target)
+          ? !category
+          : category
+      ),
+    ];
+
+    if (Number(target) === 1) {
+      setSelectCategory(saveType[0]);
+    } else {
+      if (
+        selectCategory.filter(Boolean).length === selectCategory.length - 2 &&
+        selectCategory[Number(target) - 1] === false
+      ) {
+        setSelectCategory(saveType[0]);
+      } else {
+        setSelectCategory(saveType[1]);
+      }
+    }
```

Comment:
가독성에 따라 다르겠지만, 
해당 조건문을 삼항연산자와 || 를 이용해서 반복되는 구문을 줄여주고 가독성을 좀 더 높일 수 있을 것 같습니다.
조건문은 아래와 같이 상수에 선언해 관리하실 수 있습니다.
```suggestion
    const targetCondition = Number(target) === 1
```

Reply:
```jsx
const handleCategorySubOrAll = (target, saveType) => {
    const nowCheck = selectCategory.filter(Boolean).length;
    const nowSub = selectCategory[Number(target - 1)];

    return nowCheck === selectCategory.length - 2 && !nowSub
      ? setSelectCategory(saveType[0])
      : setSelectCategory(saveType[1]);
  };

const checkCategory = (e, target) => {
    // 전첵 선택, 카테고리 선택 2가지 경우
    const saveType = [
      selectCategory.map((select, index) => (index === 0 ? !select : false)),
      selectCategory.map((select, index) =>
        index === 0 ? false : index + 1 === Number(target) ? !select : select
      ),
    ];
    const targetCondition = Number(target);
    targetCondition === 1
      ? setSelectCategory(saveType[0])
      : handleCategorySubOrAll(target, saveType);
  };
```

삼항연산자와 || 모두 사용하는 방법은 구현방법이 생각이 잘 안나서 일단 삼항연산자 활용과 else 부분 조건들을 함수로 담아서 수정해봤습니다! 삼항연산자와 || 모두 사용하는 방법은 어떤 식으로 구현해야할까요 ㅠㅠ

Reply:
첫번째 조건과 두번째 조건의 결과 값이 같으므로 아래와 같이 작성할 수 있겠네요! :) 

```js
    const targetCondition = Number(target) === 1;
    const condition =
      selectCategory.filter(Boolean).length === selectCategory.length - 2 &&
      selectCategory[Number(target) - 1] === false;

    setSelectCategory(targetCondition || condition ? saveType[0] : saveType[1]);
```

Reply:
ContextAPI 사용하면서 코드가 변경돼서 다시 수정해보겠습니다!